File: src/src_core/fmg/common/geom/SizeDouble.cs
using System;

namespace fmg.common.geom {

   public struct SizeDouble {
      public double Width, Height;

      //public SizeDouble() { width=height=0; }
      public SizeDouble(double width, double height) { Width = width; Height = height; }
      public SizeDouble(SizeDouble s) { Width = s.Width; Height = s.Height; }

      public static bool operator !=(SizeDouble s1, SizeDouble s2) { return !s1.Width.HasMinDiff(s2.Width) || !s1.Height.HasMinDiff(s2.Height); }
      public static bool operator ==(SizeDouble s1, SizeDouble s2) { return  s1.Width.HasMinDiff(s2.Width) &&  s1.Height.HasMinDiff(s2.Height); }

      public override bool Equals(object other) {
         if (ReferenceEquals(null, other))
            return false;
         return (other is SizeDouble) && (this == (SizeDouble)other);
      }
      public override int GetHashCode() {
         var bits = BitConverter.DoubleToInt64Bits(Width);
         bits ^= BitConverter.DoubleToInt64Bits(Height) * 31;
         return (((int)bits) ^ ((int)(bits >> 32)));
      }

      public override string ToString() {
         //return "{width:" + Width + ", height:" + Height + "}";
         return string.Format("{{w:{0:0.00}, h:{1:0.00}}}", Width, Height);
      }
   }

}

File: FastMines_core/src/fmg/common/geom/SizeDouble.java
package fmg.common.geom;

import java.util.Locale;

public class SizeDouble {

    public double width, height;

    public SizeDouble() { width = height = 0; }
    public SizeDouble(double width, double height) { this.width = width; this.height = height; }
    public SizeDouble(SizeDouble c) { this.width = c.width; this.height = c.height; }

    @Override
    public int hashCode() {
        long temp = Double.doubleToLongBits(height);
        int result = 31 + (int) (temp ^ (temp >>> 32));
        temp = Double.doubleToLongBits(width);
        return 31 * result + (int) (temp ^ (temp >>> 32));
    }

    @Override
    public boolean equals(Object obj) {
        if (this == obj) return true;
        if (!(obj instanceof SizeDouble))
            return false;
        SizeDouble other = (SizeDouble) obj;
        return (Double.doubleToLongBits(height) == Double.doubleToLongBits(other.height)) &&
               (Double.doubleToLongBits(width)  == Double.doubleToLongBits(other.width));
    }

    @Override
    public String toString() { return String.format(Locale.US, "{w=%.2f, h=%.2f}", width, height); }

}
